cpp/unreal_plugins/WheeledVehicle/Source/WheeledVehicle/VehiclePawn.cpp

```diff
@@ -0,0 +1,177 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#include "WheeledVehicle/VehiclePawn.h"
+
+#include <string>
+
+#include <Eigen/Dense>
+
+#include <Animation/AnimInstance.h>
+#include <Camera/CameraComponent.h>
+#include <Components/BoxComponent.h>
+#include <Components/InputComponent.h>
+#include <Components/SkeletalMeshComponent.h>
+#include <Engine/CollisionProfile.h>
+#include <UObject/ConstructorHelpers.h>
+
+#include "CoreUtils/Assert.h"
+#include "CoreUtils/Config.h"
+#include "CoreUtils/Log.h"
+#include "CoreUtils/Unreal.h"
+#include "WheeledVehicle/VehicleMovementComponent.h"
+
+AVehiclePawn::AVehiclePawn(const FObjectInitializer& object_initializer) : APawn(object_initializer)
+{
+    SP_LOG_CURRENT_FUNCTION();
+
+    if (!Config::s_initialized_) {
+        return;
+    }
+
+    ConstructorHelpers::FObjectFinder<USkeletalMesh> skeletal_mesh(*Unreal::toFString(Config::get<std::string>("WHEELED_VEHICLE.VEHICLE_PAWN.SKELETAL_MESH")));
+    SP_ASSERT(skeletal_mesh.Succeeded());
+
+    ConstructorHelpers::FClassFinder<UAnimInstance> anim_instance(*Unreal::toFString(Config::get<std::string>("WHEELED_VEHICLE.VEHICLE_PAWN.ANIM_INSTANCE")));
+    SP_ASSERT(anim_instance.Succeeded());
+
+    skeletal_mesh_component_ = CreateDefaultSubobject<USkeletalMeshComponent>(TEXT("AVehiclePawn::skeletal_mesh_component_"));
+    SP_ASSERT(skeletal_mesh_component_);
+    skeletal_mesh_component_->SetSkeletalMesh(skeletal_mesh.Object);
+    skeletal_mesh_component_->SetAnimClass(anim_instance.Class);
+    skeletal_mesh_component_->SetCollisionProfileName(UCollisionProfile::Vehicle_ProfileName);
+    skeletal_mesh_component_->BodyInstance.bSimulatePhysics = true;
+    skeletal_mesh_component_->BodyInstance.bNotifyRigidBodyCollision = true;
+    skeletal_mesh_component_->BodyInstance.bUseCCD = true;
+    skeletal_mesh_component_->bBlendPhysics = true;
+    skeletal_mesh_component_->SetGenerateOverlapEvents(true);
+    skeletal_mesh_component_->SetCanEverAffectNavigation(false);
+
+    RootComponent = skeletal_mesh_component_;
+
+    // Setup vehicle movement
+    vehicle_movement_component_ = CreateDefaultSubobject<UVehicleMovementComponent>(TEXT("AVehiclePawn::vehicle_movement_component_"));
+    SP_ASSERT(vehicle_movement_component_);
+    vehicle_movement_component_->SetIsReplicated(true); // Enable replication by default
```

Comment:
Useless comment because the information content is exactly zero. It doesn't tell me anything that I can't already tell from reading the code. Why are we doing this? Isn't replication a client-server thing? Why are we turning this on?

Reply:
Yeah, replication is required for client-server interaction in a multi-player game. Not sure why UE enables this by default in `AWheeledVehiclePawn`. I'll remove it here as it doesn't add any value.